aspnetcore/mvc/views/view-compilation.md

```diff
@@ -109,24 +109,59 @@ Build-time compilation is supplemented by runtime compilation of Razor files. Th
 The default value is `true` for:
 
 * If the app's compatibility version is set to <xref:Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_1> or earlier
-* If the app's compatibility version is set to <xref:Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_2> or later and the app is in the Development environment <xref:Microsoft.AspNetCore.Hosting.HostingEnvironmentExtensions.IsDevelopment*>. In other words, Razor files would not recompile in non-Development environment unless <xref:Microsoft.AspNetCore.Mvc.Razor.RazorViewEngineOptions.AllowRecompilingViewsOnFileChange> is explicitly set.
+* If the app's compatibility version is set to <xref:Microsoft.AspNetCore.Mvc.CompatibilityVersion.Version_2_2> or later and the app is in the Development environment <xref:Microsoft.AspNetCore.Hosting.HostingEnvironmentExtensions.IsDevelopment*>. In other words, Razor files wouldn't recompile in non-Development environment unless <xref:Microsoft.AspNetCore.Mvc.Razor.RazorViewEngineOptions.AllowRecompilingViewsOnFileChange> is explicitly set.
 
 For guidance and examples of setting the app's compatibility version, see <xref:mvc/compatibility-version>.
 
 ::: moniker-end
 
 ::: moniker range=">= aspnetcore-3.0"
 
-Runtime compilation is enabled using the `Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation` package. To enable runtime compilation, apps must:
+To enable runtime compilation for all environments and configuration modes:
 
-* Install the [Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation/) NuGet package.
-* Update the project's `Startup.ConfigureServices` method to include a call to `AddRazorRuntimeCompilation`:
+1. Install the [Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation](https://www.nuget.org/packages/Microsoft.AspNetCore.Mvc.Razor.RuntimeCompilation/) NuGet package.
 
-  ```csharp
-  services
-      .AddControllersWithViews()
-      .AddRazorRuntimeCompilation();
-  ```
+1. Update the project's `Startup.ConfigureServices` method to include a call to `AddRazorRuntimeCompilation`. For example:
+
+    ```csharp
+    public void ConfigureServices(IServiceCollection services)
+    {
+        services.AddRazorPages()
+            .AddRazorRuntimeCompilation();
+
+        // code omitted for brevity
+    }
+    ```
+
+### Conditionally enable runtime compilation
+
+To conditionally enable runtime compilation based on the environment and configuration mode:
```

Comment:
Do we want to say something about when this is useful?

Reply:
```suggestion
Runtime compilation can be conditionally enabled so that it's only available for local development. This ensures that publish output uses compiled views, is smaller in size and does not turn on file watchers in production. 

To conditionally enable runtime compilation based on the environment and configuration mode:
```

Reply:
Great idea. I'll wordsmith this a bit.